### DISCUSSION TITLE
AttributeError when using threshold_selection_tool_from_labels_table with DuckDBLinker

### DISCUSSION THREAD

**Discussion by Ahosseinzadeh723** [moj-analytical-services/splink#2193] [General]
I'm currently working with splink 3.9.13 and trying to utilize the threshold_selection_tool_from_labels_table function as demonstrated in the [documentation ](https://moj-analytical-services.github.io/splink/charts/threshold_selection_tool_from_labels_table.html?h=threshold_selection_tool_from_labels_table) found here

I've prepared my labeled table as instructed [here ](https://moj-analytical-services.github.io/splink/linker.html#splink.linker.Linker.cluster_pairwise_predictions_at_thresholds)

However, when I attempt to execute the following line of code:
```
linker.threshold_selection_tool_from_labels_table(labels_table, add_metrics=['f1'])
```
I receive the following error:
```
'DuckDBLinker' object has no attribute 'threshold_selection_tool_from_labels_table'

AttributeError: 'DuckDBLinker' object has no attribute 'threshold_selection_tool_from_labels_table'
```
This suggests that the DuckDBLinker object does not have the threshold_selection_tool_from_labels_table method. Is this functionality not supported for DuckDBLinker, or is this an issue with the library version or documentation?

Thanks!

**Comment by RobinL** [comment]
Edit 2:  The problem is that this tool appeared in 3.9.14, so you need to upgrade!  `pip install --upgrade splink` 

The following is a script that I tested as working on 3.9.14





```python
import splink.duckdb.comparison_library as cl
from splink.datasets import splink_dataset_labels, splink_datasets
from splink.duckdb.blocking_rule_library import block_on
from splink.duckdb.linker import DuckDBLinker

df = splink_datasets.fake_1000
settings = {
    "link_type": "dedupe_only",
    "blocking_rules_to_generate_predictions": [
        block_on(["first_name"]),
        block_on(["surname"]),
    ],
    "comparisons": [
        cl.jaro_winkler_at_thresholds(
            "first_name", 0.9, term_frequency_adjustments=True
        ),
        cl.exact_match("surname"),
        cl.levenshtein_at_thresholds("dob"),
        cl.exact_match("city", term_frequency_adjustments=True),
        cl.exact_match("email"),
    ],
}

linker = DuckDBLinker(df, settings)

linker.estimate_probability_two_random_records_match(
    [block_on(["first_name", "surname", "dob"])], recall=0.7
)

linker.estimate_parameters_using_expectation_maximisation(block_on(["first_name"]))

linker.estimate_parameters_using_expectation_maximisation(block_on(["surname"]))

labels_splink_df = linker.register_labels_table(splink_dataset_labels.fake_1000_labels)
linker.threshold_selection_tool_from_labels_table(labels_splink_df)

```
